common/src/main/java/hunternif/mc/impl/atlas/core/scanning/TileDetectorBase.java

```diff
@@ -41,36 +47,54 @@ public class TileDetectorBase implements ITileDetector {
     private static final int ravineMinDepth = 7;
 
-    /**
-     * Set to true for biome IDs that return true for BiomeDictionary.isBiomeOfType(WATER)
-     */
-    private static final Set<Identifier> waterBiomes = new HashSet<>();
-    /**
-     * Set to true for biome IDs that return true for BiomeDictionary.isBiomeOfType(BEACH)
-     */
-    private static final Set<Identifier> beachBiomes = new HashSet<>();
-
-    private static final Set<Identifier> swampBiomes = new HashSet<>();
-
-    /**
-     * Scan all registered biomes to mark biomes of certain types that will be
-     * given higher priority when identifying mean biome ID for a chunk.
-     * (Currently WATER, BEACH and SWAMP)
-     */
-    public static void scanBiomeTypes() {
-        for (Biome biome : BuiltinRegistries.BIOME) {
-            switch (biome.getCategory()) {
-                case BEACH -> beachBiomes.add(BuiltinRegistries.BIOME.getId(biome));
-                case RIVER, OCEAN -> waterBiomes.add(BuiltinRegistries.BIOME.getId(biome));
-                case SWAMP -> swampBiomes.add(BuiltinRegistries.BIOME.getId(biome));
-            }
-        }
+//  Seems like init-time biome classification isn't a thing anymore, so, can't cache these, at least not out of world.
+//    /**
+//     * Set to true for biome IDs that return true for BiomeDictionary.isBiomeOfType(WATER)
+//     */
+//    private static final Set<Identifier> waterBiomes = new HashSet<>();
+//    /**
+//     * Set to true for biome IDs that return true for BiomeDictionary.isBiomeOfType(BEACH)
+//     */
+//    private static final Set<Identifier> beachBiomes = new HashSet<>();
+//
+//    private static final Set<Identifier> swampBiomes = new HashSet<>();
+//
+//    /**
+//     * Scan all registered biomes to mark biomes of certain types that will be
+//     * given higher priority when identifying mean biome ID for a chunk.
+//     * (Currently WATER, BEACH and SWAMP)
+//     */
+//    public static void scanBiomeTypes() {
+//        for (Biome biome : BuiltinRegistries.BIOME) {
+//            switch (biome.getCategory()) {
+//                case BEACH -> beachBiomes.add(BuiltinRegistries.BIOME.getId(biome));
+//                case RIVER, OCEAN -> waterBiomes.add(BuiltinRegistries.BIOME.getId(biome));
+//                case SWAMP -> swampBiomes.add(BuiltinRegistries.BIOME.getId(biome));
+//            }
+//        }
+//    }
+//
+//    int priorityForBiome(Identifier biome) {
+//        if (waterBiomes.contains(biome)) {
+//            return 4;
+//        } else if (beachBiomes.contains(biome)) {
+//            return 3;
+//        } else {
+//            return 1;
+//        }
+//    }
+
+    @ExpectPlatform
+    static private boolean hasSwampWater(RegistryEntry<Biome> biomeTag) {
+        throw new AssertionError("Not implemented");
     }
 
-    int priorityForBiome(Identifier biome) {
-        if (waterBiomes.contains(biome)) {
+    public static int priorityForBiome(World world, Biome biome) {
```

Comment:
This can be private

Reply:
used in TileDetectorNether! Updated to package-private in a61ff06